gemrb/plugins/BAMImporter/BAMImporter.cpp

```diff
@@ -151,22 +252,41 @@ std::vector<BAMImporter::index_t> BAMImporter::CacheFLT()
 
 AnimationFactory* BAMImporter::GetAnimationFactory(const ResRef &resref, bool allowCompression)
 {
-	str->Seek( DataStart, GEM_STREAM_START );
-	strpos_t length = str->Remains();
-	if (length == 0) return nullptr;
-	
-	auto FLT = CacheFLT();
-	uint8_t *data = (uint8_t*)malloc(length);
-	str->Read(data, length);
-
 	std::vector<Holder<Sprite2D>> animframes;
-	for (const auto& frameInfo : frames) {
-		bool RLECompressed = allowCompression && frameInfo.RLE;
-		animframes.push_back(GetFrameInternal(frameInfo, RLECompressed, data - DataStart));
-	}
-	free(data);
 
-	return new AnimationFactory(resref, std::move(animframes), cycles, std::move(FLT));
+	if (version == BAMVersion::V1) {
+		str->Seek( DataStart, GEM_STREAM_START );
+		strpos_t length = str->Remains();
+		if (length == 0) return nullptr;
+
+		auto FLT = CacheFLT();
+		uint8_t *data = (uint8_t*)malloc(length);
+		str->Read(data, length);
+
+		for (const auto& frameInfo : frames) {
+			bool RLECompressed = allowCompression && frameInfo.RLE;
+			animframes.push_back(GetFrameInternal(frameInfo, RLECompressed, data - DataStart));
+		}
+		free(data);
+
+		return new AnimationFactory(resref, std::move(animframes), cycles, std::move(FLT));
+	} else {
+		std::vector<index_t> FLT;
+		FLT.resize(frames.size());
```

Comment:
Same as `std::vector<index_t> FLT(frames.size());` and the commented code below should be removed.

Reply:
Force of habit, as without caution the ctor `std::vector` may interpret the size as an item to incorporate.